src/decorators/OrmRepository.ts

```diff
@@ -1,21 +1,179 @@
+ * ```ts
+ * class Sample {
+ *   // constructor injection
+ *   constructor(
+ *     \@OrmRepository(User, "test-conn")
+ *      private userRepository: Repository<User>,
+ *   ) {}
+ * 
+ *   // property injection
+ *  \@OrmRepository(User, "test-conn")
+ *   userRepository: Repository<User>;
+ * }
+ * ```
+ */
+export function OrmRepository(entityType: Function, connectionName: string): ParamOrPropDecorator;
+
+export function OrmRepository(entityTypeOrConnectionName?: Function|string, paramConnectionName = "default"): ParamOrPropDecorator {
+    return (object: object, propertyName: string, index?: number) => {
+        let entityType: Function|undefined;
+        let connectionName: string;
+        let repositoryType: Function;
+
+        // handle first parameter overload
+        connectionName = paramConnectionName;
+        if (typeof entityTypeOrConnectionName === "string") {
+            connectionName = entityTypeOrConnectionName;
+        } else if (typeof entityTypeOrConnectionName === "function") {
+            entityType = entityTypeOrConnectionName;
+        }
+
+        // if the decorator has been aplied to parameter (constructor injection)
+        if (index) {
+            const paramTypes: Function[] | undefined = Reflect.getOwnMetadata("design:paramtypes", object, propertyName);
+            if (!paramTypes || !paramTypes[index]) {
+                throw new Error(
+                    `Cannot get reflected type for a "${propertyName}" method's ${index + 1}. parameter of ${object.constructor.name} class. ` +
+                    `Make sure you have turned on an "emitDecoratorMetadata": true, option in tsconfig.json. ` +
+                    `and that you have imported "reflect-metadata" on top of the main entry file in your application.` +
+                    `And make sure that you have annotated the property type correctly with: ` +
+                    `Repository, MongoRepository, TreeRepository or custom repository class type.`
+                );
+            }
+            repositoryType = paramTypes[index];
+        }
+        // if the parameter has been aplied to class property
+        else {
+            const propertyType: Function | undefined = Reflect.getOwnMetadata("design:type", object, propertyName);
+            if (!propertyType) {
+                throw new Error(
+                    `Cannot get reflected type for a property "${propertyName}" of ${object.constructor.name} class. ` +
+                    `Make sure you have turned on an "emitDecoratorMetadata": true, option in tsconfig.json ` +
+                    `and that you have imported "reflect-metadata" on top of the main entry file in your application.` +
+                    `And make sure that you have annotated the property type correctly with: ` +
+                    `Repository, MongoRepository, TreeRepository or custom repository class type.`
+                );
+            }
+            repositoryType = propertyType;
```

Comment:
If Im not wrong `connection.getRepository` always returns you `MongoRepository` for mongodb, `TreeRepository` for tree entities, and regular `Repository` for regular entities, so I think you don't really need "repository type" to make different calls to connection.

Reply:
If you are sure, I can do this, but the same switch-case already exist in `@TransactionRepository` decorator code 😉